src/data/blob.rs

```diff
@@ -0,0 +1,420 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use crate::{utils, Error, PublicKey, XorName};
+use bincode::serialized_size;
+use multibase::Decodable;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    fmt::{self, Debug, Formatter},
+    u64,
+};
+use tiny_keccak;
+
+/// Maximum allowed size for a serialised blob (ID) to grow to
+pub const MAX_BLOB_SIZE_IN_BYTES: u64 = 1024 * 1024 + 10 * 1024;
+
+/// PrivateBlob: an immutable chunk of data which can be deleted. Can only be fetched
+/// by the listed owner.
+#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone)]
+pub struct PrivateBlob {
+    /// Address.
+    address: Address,
+    /// Contained data.
+    value: Vec<u8>,
+    /// Contains a set of owners of this data. DataHandlers enforce that a DELETE or OWNED-GET type
+    /// of request is coming from the ClientHandler of the owners.
+    owner: PublicKey,
+}
+
+impl PrivateBlob {
+    /// Creates a new instance of `PrivateBlob`
+    pub fn new(value: Vec<u8>, owner: PublicKey) -> Self {
+        let hash_of_value = tiny_keccak::sha3_256(&value);
+        let serialised_contents = utils::serialise(&(hash_of_value, &owner));
+        let address = Address::Private(XorName(tiny_keccak::sha3_256(&serialised_contents)));
+
+        Self {
+            address,
+            value,
+            owner,
+        }
+    }
+
+    /// Returns the value.
+    pub fn value(&self) -> &Vec<u8> {
+        &self.value
+    }
+
+    /// Returns the set of owners.
+    pub fn is_owner(&self, user: PublicKey) -> bool {
+        self.owner == user
+    }
+
+    /// Returns the set of owners.
+    pub fn owner(&self) -> &PublicKey {
+        &self.owner
+    }
+
+    /// Returns the address.
+    pub fn address(&self) -> &Address {
+        &self.address
+    }
+
+    /// Returns the name.
+    pub fn name(&self) -> &XorName {
+        self.address.name()
+    }
+
+    /// Returns size of contained value.
+    pub fn payload_size(&self) -> usize {
+        self.value.len()
+    }
+
+    /// Returns size of this BlobData after serialisation.
+    pub fn serialised_size(&self) -> u64 {
+        serialized_size(self).unwrap_or(u64::MAX)
+    }
+
+    /// Return true if the size is valid
```

Comment:
```suggestion
    /// Returns true if the size is valid.
```
Just to keep it consistent with the rest of the documentation :)

Reply:
Yep, this is part of docs/code that has not been modified in this work.
Maybe hard to see when there has been renaming all over.

Reply:
I like to keep verbs in the third person to be consistent with the Rust standard library documentation